src/charm.py

```diff
@@ -354,13 +363,29 @@ def push_tls_files_to_workload(self) -> None:
                 cert,
                 0o400,
             )
-
-        self.update_config()
+        if update_config:
+            self.update_config()
```

Comment:
If called from `_on_pebble_ready()`, the file may not have been rendered yet and will be rendered anyway by `_on_start()`